emily/handler/src/api/routes/mod.rs

```diff
@@ -43,3 +47,41 @@ pub fn routes(
         // Convert reply to tuple to that more routes can be added to the returned filter.
         .map(|reply| (reply,))
 }
+
+/// This function sets up the routes expecting the AWS stage to be passed in as the very
+/// first segment of the path. AWS does this by default, and it's not something we can
+/// change.
+pub fn routes_with_stage_prefix(
+    context: EmilyContext,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = warp::Rejection> + Clone {
+    // Get the AWS Stage name and then ignore it, but print it in the logs if
+    // we're in debug mode.
+    warp::path::param::<String>()
+        .and(routes(context))
+        .map(|stage, reply| {
+            debug!("AWS stage: {}", stage);
+            (reply,)
+        })
+}
+
+/// A verbose route that will return a 404 with the full path and peeked path.
+///
```

Comment:
just checking if this extra line is intentional

Reply:
It was intentional, do you like it?